src/SymmetricPositiveDefinite.jl

```diff
@@ -444,3 +444,25 @@ function check_tangent_vector(M::SymmetricPositiveDefinite{N},x,v; kwargs...) wh
     return nothing
 end
 check_tangent_vector(M::MetricManifold{SymmetricPositiveDefinite{N},LogCholeskyMetric},x,v; kwargs...) where N = check_tangent_vector(M.manifold,x,v;kwargs...)
+
+"""
+    mean(M::SymmetricPositiveDefinite, x::AbstractVector[, w::AbstractWeights]; kwargs...)
+
+Compute the Riemannian
+[`mean`](@ref mean(::Manifold, ::AbstractVector, ::AbstractVector, ::GeodesicInterpolation))
+of `x` using [`GeodesicInterpolation`](@ref).
+"""
+mean(::SymmetricPositiveDefinite, args...)
+
+mean!(M::SymmetricPositiveDefinite, y, x::AbstractVector, w::AbstractVector; kwargs...) =
+    mean!(M, y, x, w, GeodesicInterpolation(); kwargs...)
```

Comment:
The most efficient one should be cyclic proximal point, I think. SPDs are a Hadamard manifold, so the distance is convex globally and the results from Bacak apply.

Reply:
We don't currently have a `CyclicProximalPoint` version of `mean!`.

Reply:
Oh. I will provide that in an upcoming PR then, there is just a slight change (the parameter t changes) compared to the median.

Reply:
And of course, we should probably benchmark the two techniques. The geodesic interpolation approach performs exactly `n-1` `retract!` and `inverse_retract!` calls for `n=length(x)`. The Cyclic proximal point implementation of `median!` requires `m * n` `retract!` and `inverse_retract!` for `m≥2`, where `m` is the number of iterations performed until convergence is detected. If the `mean!` version of cyclic proximal point has similar structure, I don't see how it could be more efficient. It could, however, be more accurate.